### PR TITLE
[Feature]: Install your previous dotfiles from your dotfiles repository

### PR DESCRIPTION
**Pull request by gtrabanco**
# Description
This adds a recover file to install your dotfiles.

# Motivation
Make simpler the restoration of your dotfiles

# Testing
For me this works quite right. It will receive updates soon when more features were added. It would be nice make some more testing overall with Linux distros because the only test I did was in my intel osx.

# Comment
If you remember this PR was open with Help needed. The problem was if any link fail in dotbot it was returning exit code error which made the program stop due to set option (`set -e`). It should be solved now.

# Task

- [x] Better handling of error to avoid `set +e`
- [x] Test in MBP x86_64 Big Sur

### CODE REVIEW

**Comment by gtrabanco** [comment]
This needs testing

**Comment by gtrabanco** [comment]
The test by my part were done in some special circunstances.

I am not using current DOTLY version. I am using DOTLY from my repository which uses ssh git urls for dotly repository because I use to make changes in my dotly repository as you can see with all pending PRs for a long time.
This needs testing from a person which uses official DOTLY version. A fail in the script when executing `dot self install` could be expected. In installation dotbot will fail because PATH is not well defined for newly installed brew gnu tools (in osx only) and this makes dot symlinks apply to exit with fail code. This is managed in PR #142.

It should not fail any other sutff.

**Comment by gtrabanco** [comment]
Tested in brand new install. So I have this pending Task to fix completely the install:

# Tasks

- [ ] Install Rust fail when run `dot self install`
- [ ] Install docpars after install rust manually and re-run the script fail. Check if `$PATH` was updated with cargo `PATH`.

# Important Question

@rgomezcasas Why rust is not installed with Package manager or at least try to install it with package manager instead of using official script to install?

**Comment by gtrabanco** [comment]
Probably this wold be better in install script and select if you want to recover your dotfiles or create a new dotfiles? What do you think @rgomezcasas ?

File: dotfiles_template/README.md [hunk 1]
```diff
@@ -2,12 +2,28 @@
...
+
+1. Execute
+```bash
+bash <(curl -s https://raw.githubusercontent.com/gtrabanco/dotly/feature/easy-recovery)
+```
+<!-- Replace in Pull Request
+```bash
+bash <(curl -s https://raw.githubusercontent.com/codelytv/dotly/master/easy-recovery)
+```
+-->
```
Comment (rgomezcasas) [inline dotfiles_template/README.md:25]:
```suggestion
1. Execute
```bash
bash <(curl -s https://raw.githubusercontent.com/codelytv/dotly/master/easy-recovery)
```

```

**Review by rgomezcasas — APPROVED** [review]
Oleeeee!! 👏👏👏👏 Thanks a lot!!

We're going to do some changes, but it's 👌

**Pull request merged**